### ISSUE TITLE
Chat

### ISSUE THREAD

**Issue by Fr0go1** [TKazer/CS2_External#4]
How do i use it?

**Comment by Fr0go1** [comment]
i found out how, how do i turn off team esp

**Comment by Fr0go1** [comment]
and make it change color if visible and turn off the box

**Comment by dasstf** [comment]
have u solved the problem？  i am concerned the same you conveyed

**Comment by Fr0go1** [comment]
> have u solved the problem？ i am concerned the same you conveyed

no

**Comment by Fr0go1** [comment]
![image](https://github.com/TKazer/CS2_External/assets/112802612/e7e988cc-fd00-48c1-b6c9-2b5dfaf9021e)
i removed the box, i set the alpha to 0

**Comment by xleakfraud** [comment]
Can someone tell me how to open the external and use it

**Comment by Fr0go1** [comment]
> Can someone tell me how to open the external and use it

just open it when cs2 is open

**Comment by Fr0go1** [comment]
im trying to make a head box but it changes size based on distance (i know why but if i do it any other way it doesnt work)

`	void ShowHeadBox(const CEntity& Entity, ImColor Color, float Thickness)
	{
		BoneJointPos Head = Entity.GetBone().BonePosList[BONEINDEX::head];

		if (Head.IsVisible)
		{
			// Define the size of the box around the head
			ImVec2 Size = ImVec2(20.0f, 20.0f); // Adjust the size as needed

			// Calculate the minimum and maximum points for the box
			ImVec2 Min = ImVec2(Head.ScreenPos.x - Size.x / 2, Head.ScreenPos.y - Size.y / 2);
			ImVec2 Max = ImVec2(Head.ScreenPos.x + Size.x / 2, Head.ScreenPos.y + Size.y / 2);

			// Draw the rectangle around the head
			Gui.Rectangle(Min, ImVec2(Max.x - Min.x, Max.y - Min.y), Color, Thickness);
		}
	}
`

**Comment by xleakfraud** [comment]
> ![image](https://user-images.githubusercontent.com/112802612/271775231-e7e988cc-fd00-48c1-b6c9-2b5dfaf9021e.png) i removed the box, i set the alpha to 0

Can u tell me how?

**Comment by Fr0go1** [comment]
> > ![image](https://user-images.githubusercontent.com/112802612/271775231-e7e988cc-fd00-48c1-b6c9-2b5dfaf9021e.png) i removed the box, i set the alpha to 0
> 
> Can u tell me how?

go to cheats.cpp and make the last number in the imcolor 0 
![image](https://github.com/TKazer/CS2_External/assets/112802612/532018be-9a31-407b-a53f-0f7aa543c706)

**Comment by xleakfraud** [comment]
thanks , do u know how to remove the team esp?

**Comment by Fr0go1** [comment]
> thanks , do u know how to remove the team esp?

my guess it the dev who made this was already working on it

![image](https://github.com/TKazer/CS2_External/assets/112802612/c7f58e81-2a6a-45c8-b9ce-afe28054742b)

this is checking things before it continues to render so the commented out text (the green) is most likely a team check but teamid is not yet implemented by the dev

**Comment by Fr0go1** [comment]
i want to make a gui but im not sure how os-imgui works and how different it is from imgui

**Comment by Fr0go1** [comment]
https://www.unknowncheats.me/forum/counter-strike-2-a/591548-external-visibility-check.html

https://github.com/bananasss00/Perfect-Visible-Check-External-

found some shit to help make a visibility check

**Comment by xleakfraud** [comment]
btw why doesnt it work in real games with people

**Comment by xleakfraud** [comment]
nevermind it works
btw frogo do u ave discord?
if u have can u add me : xleakfraud

**Comment by jintan1994** [comment]
@Fr0go1 can you tell me how to use this ? pelase i have no idea

**Comment by Fr0go1** [comment]
> @Fr0go1 can you tell me how to use this ? pelase i have no idea

just open it when csgo is open 

but first you need to compile it with visual studio

**Comment by TKazer** [comment]
> i want to make a gui but im not sure how os-imgui works and how different it is from imgui

You just need to use Imgui api in Cheats function, os-imgui is only the library to use imgui easily.

**Comment by Fr0go1** [comment]
> > i want to make a gui but im not sure how os-imgui works and how different it is from imgui
> 
> You just need to use Imgui api in Cheats function, os-imgui is only the library to use imgui easily.

so its the same? also can you please add team check

**Comment by TKazer** [comment]
> > > i want to make a gui but im not sure how os-imgui works and how different it is from imgui
> > 
> > 
> > You just need to use Imgui api in Cheats function, os-imgui is only the library to use imgui easily.
> 
> so its the same? also can you please add team check

it has already exist, I forgot to cancel the code comment in cheats function, you just need to add `if (Entity.Controller.TeamID==LocalEntity.Controller.TeamID) continue;`

**Comment by TKazer** [comment]
> > > > i want to make a gui but im not sure how os-imgui works and how different it is from imgui
> > > 
> > > 
> > > You just need to use Imgui api in Cheats function, os-imgui is only the library to use imgui easily.
> > 
> > 
> > so its the same? also can you please add team check
> 
> it has already exist, I forgot to cancel the code comment in cheats function, you just need to add `if (Entity.Controller.TeamID==LocalEntity.Controller.TeamID) continue;`

I has updated the cheats.cpp just now.

**Comment by Fr0go1** [comment]
thanks

> > > > > i want to make a gui but im not sure how os-imgui works and how different it is from imgui
> > > > 
> > > > 
> > > > You just need to use Imgui api in Cheats function, os-imgui is only the library to use imgui easily.
> > > 
> > > 
> > > so its the same? also can you please add team check
> > 
> > 
> > it has already exist, I forgot to cancel the code comment in cheats function, you just need to add `if (Entity.Controller.TeamID==LocalEntity.Controller.TeamID) continue;`
> 
> I has updated the cheats.cpp just now.

thansk

**Comment by Fr0go1** [comment]
is visible check possible? like it changes color depending on if visible

**Comment by TKazer** [comment]
> is visible check possible? like it changes color depending on if visible

It will be add soon. With weapon esp and glow.

**Comment by Fr0go1** [comment]
also how would i make a toggle for box esp

**Comment by Fr0go1** [comment]
> > is visible check possible? like it changes color depending on if visible
> 
> It will be add soon. With weapon esp and glow.

dont add glow it is dtc because it writes to cs2

**Comment by Fr0go1** [comment]
chams is better but is impossible with external

**Comment by Fr0go1** [comment]
how do i make a head box 

	void ShowHeadBox(const CEntity& Entity, ImColor Color, float Thickness)
	{
		BoneJointPos Head = Entity.GetBone().BonePosList[BONEINDEX::head];

		if (Head.IsVisible)
		{
			ImVec2 Size = ImVec2(15.0f, 15.0f);

			ImVec2 Min = ImVec2(Head.ScreenPos.x - Size.x / 2, Head.ScreenPos.y - Size.y / 2);
			ImVec2 Max = ImVec2(Head.ScreenPos.x + Size.x / 2, Head.ScreenPos.y + Size.y / 2);

			Gui.Rectangle(Min, ImVec2(Max.x - Min.x, Max.y - Min.y), Color, Thickness);
		}
	}

**Comment by kura6789** [comment]
> how do i make a head box
> 
> ```
> void ShowHeadBox(const CEntity& Entity, ImColor Color, float Thickness)
> {
> 	BoneJointPos Head = Entity.GetBone().BonePosList[BONEINDEX::head];
> 
> 	if (Head.IsVisible)
> 	{
> 		ImVec2 Size = ImVec2(15.0f, 15.0f);
> 
> 		ImVec2 Min = ImVec2(Head.ScreenPos.x - Size.x / 2, Head.ScreenPos.y - Size.y / 2);
> 		ImVec2 Max = ImVec2(Head.ScreenPos.x + Size.x / 2, Head.ScreenPos.y + Size.y / 2);
> 
> 		Gui.Rectangle(Min, ImVec2(Max.x - Min.x, Max.y - Min.y), Color, Thickness);
> 	}
> }
> ```

thsnks but how to make them smaller if they are far away :D ?

**Comment by Fr0go1** [comment]
> > how do i make a head box
> > ```
> > void ShowHeadBox(const CEntity& Entity, ImColor Color, float Thickness)
> > {
> > 	BoneJointPos Head = Entity.GetBone().BonePosList[BONEINDEX::head];
> > 
> > 	if (Head.IsVisible)
> > 	{
> > 		ImVec2 Size = ImVec2(15.0f, 15.0f);
> > 
> > 		ImVec2 Min = ImVec2(Head.ScreenPos.x - Size.x / 2, Head.ScreenPos.y - Size.y / 2);
> > 		ImVec2 Max = ImVec2(Head.ScreenPos.x + Size.x / 2, Head.ScreenPos.y + Size.y / 2);
> > 
> > 		Gui.Rectangle(Min, ImVec2(Max.x - Min.x, Max.y - Min.y), Color, Thickness);
> > 	}
> > }
> > ```
> 
> thsnks but how to make them smaller if they are far away :D ?

i dont know i sent that code wondering if someone else can fix it

**Comment by TKazer** [comment]
```
void ShowHeadBox(const CEntity& Entity, ImColor Color, float Thickness)
{
	BoneJointPos Head = Entity.GetBone().BonePosList[BONEINDEX::head];

	if (Head.IsVisible)
	{
		Vec2 Size;
		Size.y = Entity.Pawn.ScreenPos.y - Head.ScreenPos.y;
		Size.x = Size.y * 0.64;

		Vec2 Pos = ImVec2(Entity.Pawn.ScreenPos.x - Size.x / 2, Head.ScreenPos.y);

		Gui.Rectangle(Pos, Size, Color, Thickness);
	}
}
 ```

**Comment by Fr0go1** [comment]
> ```
> void ShowHeadBox(const CEntity& Entity, ImColor Color, float Thickness)
> {
>    BoneJointPos Head = Entity.GetBone().BonePosList[BONEINDEX::head];
> 
>    if (Head.IsVisible)
>    {
>    	Vec2 Size;
>    	Size.y = Entity.Pawn.ScreenPos.y - Head.ScreenPos.y;
>    	Size.x = Size.y * 0.64;
> 
>    	Vec2 Pos = ImVec2(Entity.Pawn.ScreenPos.x - Size.x / 2, Head.ScreenPos.y);
> 
>    	Gui.Rectangle(Pos, Size, Color, Thickness);
>    }
> }
> ```

thanks

**Comment by Fr0go1** [comment]
@TKazer can you please make a non dynamic box esp (doesnt change size when they walk) and add some features from this (like health text and the simple bar) https://github.com/UnnamedZ03/CS2-external-base

**Comment by TKazer** [comment]
> @TKazer can you please make a non dynamic box esp (doesnt change size when they walk) and add some features from this (like health text and the simple bar) https://github.com/UnnamedZ03/CS2-external-base

that you need to make these by yourself, changing style isn’t in my plans.

**Comment by Fr0go1** [comment]
> > @TKazer can you please make a non dynamic box esp (doesnt change size when they walk) and add some features from this (like health text and the simple bar) https://github.com/UnnamedZ03/CS2-external-base
> 
> that you need to make these by yourself, changing style isn’t in my plans.

i know i didnt mean for oyu to change style i meant you change that style to yours

**Comment by Fr0go1** [comment]
also weaponesp isnt rendered so it wont work

**Comment by dasstf** [comment]
> also weaponesp isnt rendered so it wont work

if your code works？  my code report mistakes c2665 about the undentifier of the "Rect“， tried a lot but no difference

**Comment by Fr0go1** [comment]
> > also weaponesp isnt rendered so it wont work
> 
> if your code works？ my code report mistakes c2665 about the undentifier of the "Rect“， tried a lot but no difference

delete the folder your code is in and download the code again from here and dont edit it

**Comment by Fr0go1** [comment]
> > @TKazer can you please make a non dynamic box esp (doesnt change size when they walk) and add some features from this (like health text and the simple bar) https://github.com/UnnamedZ03/CS2-external-base
> 
> that you need to make these by yourself, changing style isn’t in my plans.

let me rephrase, can you please add health text and distance text

**Comment by Fr0go1** [comment]
i got chatgpt to add humanization but the "*" on this line  Vec3 p = p0 ***** uuu + p1 ***** (3 * uu * t) + p2 ***** (3 * u * tt) + p3 ***** ttt; is messed up

#pragma once
#define _USE_MATH_DEFINES
#include <math.h>
#include "Game.h"
#include "Entity.h"
#include <iostream>

namespace AimControl
{
    static int HotKey = VK_LMENU;   // Hotkey
    static float AimRange = 300;    // Aim range
    static float Smooth = 0.7;      // Smoothing factor

    // Bezier interpolation function with more control points
    inline Vec3 BezierInterpolation(const Vec3& p0, const Vec3& p1, const Vec3& p2, const Vec3& p3, float t)
    {
        float u = 1 - t;
        float tt = t * t;
        float uu = u * u;
        float uuu = uu * u;
        float ttt = tt * t;

        Vec3 p = p0 * uuu + p1 * (3 * uu * t) + p2 * (3 * u * tt) + p3 * ttt;

        return p;
    }

    inline void AimBot(const CEntity& Local, Vec3 LocalPos, Vec3 AimPos, float t)
    {
        float Yaw, Pitch;
        float Distance;
        Vec3 OppPos;

        // Use more control points for smoother interpolation
        Vec3 controlPoint1 = LocalPos + (AimPos - LocalPos) * 0.3;
        Vec3 controlPoint2 = LocalPos + (AimPos - LocalPos) * 0.6;
        Vec3 controlPoint3 = AimPos;

        OppPos = BezierInterpolation(LocalPos, controlPoint1, controlPoint2, controlPoint3, t) - LocalPos;

        Distance = sqrt(pow(OppPos.x, 2) + pow(OppPos.y, 2));

        Yaw = atan2f(OppPos.y, OppPos.x) * 180 / M_PI - Local.Pawn.ViewAngle.y;
        Pitch = -atan(OppPos.z / Distance) * 180 / M_PI - Local.Pawn.ViewAngle.x;
        Yaw = Yaw * Smooth + Local.Pawn.ViewAngle.y;
        Pitch = Pitch * Smooth + Local.Pawn.ViewAngle.x;

        // Recoil control
        if (Local.Pawn.ShotsFired > 1)
        {
            Yaw = Yaw - Local.Pawn.AimPunchAngle.y * 1.2;
            Pitch = Pitch - Local.Pawn.AimPunchAngle.x * 1.4;
        }

        gGame.SetViewAngle(Yaw, Pitch);
    }
}

**Comment by kura6789** [comment]
> ```
> void ShowHeadBox(const CEntity& Entity, ImColor Color, float Thickness)
> {
>    BoneJointPos Head = Entity.GetBone().BonePosList[BONEINDEX::head];
> 
>    if (Head.IsVisible)
>    {
>    	Vec2 Size;
>    	Size.y = Entity.Pawn.ScreenPos.y - Head.ScreenPos.y;
>    	Size.x = Size.y * 0.64;
> 
>    	Vec2 Pos = ImVec2(Entity.Pawn.ScreenPos.x - Size.x / 2, Head.ScreenPos.y);
> 
>    	Gui.Rectangle(Pos, Size, Color, Thickness);
>    }
> }
> ```

is it working for you? i got same box size as normal box ESP

**Comment by Fr0go1** [comment]
@TKazer im uploading my source to git rn, it has a config saver and some changes so you can pick and add to yours

**Comment by Fr0go1** [comment]
https://github.com/Fr0go1/cs2

**Comment by TKazer** [comment]
> https://github.com/Fr0go1/cs2

you can download the new vertion, i has added a menu and some new funtions.
and i'll add a new configSystem soon. thank you for sharing.

**Comment by Fr0go1** [comment]
> > https://github.com/Fr0go1/cs2
> 
> you can download the new vertion, i has added a menu and some new funtions. and i'll add a new configSystem soon. thank you for sharing.

yeah i downloaded the new version and am going to start pasting things from mine

![image](https://github.com/TKazer/CS2_External/assets/112802612/609a715a-317a-4c94-a586-0e68d0a5ed89)
this is my todo wish list so i will slowly implement it (not today cuz i just went on a huge bike ride through the tracks so i just wanna relax)

**Comment by Fr0go1** [comment]
vmprotect just makes the exe a little more ud and simple health bar i meant vertical and you added so thats great

**Comment by Fr0go1** [comment]
can you make a better radar like we can move it

**Comment by TKazer** [comment]
> > > ```
> > > void ShowHeadBox(const CEntity& Entity, ImColor Color, float Thickness)
> > > {
> > >    BoneJointPos Head = Entity.GetBone().BonePosList[BONEINDEX::head];
> > > 
> > >    if (Head.IsVisible)
> > >    {
> > >    	Vec2 Size;
> > >    	Size.y = Entity.Pawn.ScreenPos.y - Head.ScreenPos.y;
> > >    	Size.x = Size.y * 0.64;
> > > 
> > >    	Vec2 Pos = ImVec2(Entity.Pawn.ScreenPos.x - Size.x / 2, Head.ScreenPos.y);
> > > 
> > >    	Gui.Rectangle(Pos, Size, Color, Thickness);
> > >    }
> > > }
> > > ```
> > 
> > 
> > is it working for you? i got same box size as normal box ESP
> 
> Can u help with this? @Fr0go1 in ur source it isnt fixed too

this type of box has already been added to project.

**Comment by kura6789** [comment]
Head box? Can't see this im cheata.cpp and drawing

**Comment by TKazer** [comment]
Get2DBox() is the head box.

**Comment by Fr0go1** [comment]
> Get2DBox() is the head box.

he means just a box only on the head

**Comment by Fr0go1** [comment]
also can you make the ui like this

https://github.com/RequestFX/ImGUI-Advanced-Cheat-Menu
https://www.unknowncheats.me/forum/c-and-c-/505940-advanced-imgui-cheat-menu.html

**Comment by TKazer** [comment]
> also can you make the ui like this
> 
> https://github.com/RequestFX/ImGUI-Advanced-Cheat-Menu https://www.unknowncheats.me/forum/c-and-c-/505940-advanced-imgui-cheat-menu.html

no, i wont design a menu for this project, bcz this project just use as reference and copy, so some private design should be make by self.

**Comment by Fr0go1** [comment]
> > also can you make the ui like this
> > https://github.com/RequestFX/ImGUI-Advanced-Cheat-Menu https://www.unknowncheats.me/forum/c-and-c-/505940-advanced-imgui-cheat-menu.html
> 
> no, i wont design a menu for this project, bcz this project just use as reference and copy, so some private design should be make by self.

ok

**Comment by Fr0go1** [comment]
if someone were to make it would you use it?

**Comment by kura6789** [comment]
> Get2DBox() is the head box.

Is this drawing head box? So i need to check it

**Comment by exscreamo** [comment]
how to enable ESP box when spectating a teammate?

**Comment by Fr0go1** [comment]
i made a few changes you can use the code or the exe idm https://github.com/Fr0go1/Aeonix-Cs2
